### PR TITLE
[macOS] compute job accounting info

### PR DESCRIPTION
**Pull request by almiliMSFT**
This PR adds job accounting information for process pips when executing on macOS.

This info includes processor system and user time, as well as peak memory usage.

This is achieved by probing the process usage information every 2 seconds and aggregating the results.

Computing this information is currently disabled by default, and can be enabled by passing `/kextMeasureProcessCpuTimes`.   This default is likely to change soon.

### CODE REVIEW

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs [hunk 1]
```diff
@@ -229,13 +229,14 @@ public virtual async Task<SandboxedProcessResult> GetResultAsync()
 
             var reportFileAccesses = ProcessInfo.FileAccessManifest?.ReportFileAccesses == true;
             var fileAccesses = reportFileAccesses ? (reports?.FileAccesses ?? s_emptyFileAccessesSet) : null;
-
+            var processTimes = GetProcessTimes();
```
Comment (almiliMSFT) [inline Public/Src/Engine/Processes/UnSandboxedProcess.cs]:
```suggestion
```

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs [hunk 2]
```diff
@@ -247,7 +248,7 @@ public virtual async Task<SandboxedProcessResult> GetResultAsync()
                 MessageProcessingFailure            = reports?.MessageProcessingFailure,
                 DumpCreationException               = m_dumpCreationException,
                 DumpFileDirectory                   = ProcessInfo.TimeoutDumpDirectory,
-                PrimaryProcessTimes                 = GetProcessTimes(),
+                PrimaryProcessTimes                 = processTimes,
```
Comment (almiliMSFT) [inline Public/Src/Engine/Processes/UnSandboxedProcess.cs]:
```suggestion
                PrimaryProcessTimes                 = GetProcessTimes(),
```

File: Public/Src/Engine/Processes/SandboxedProcess.MacKext.cs [hunk 3]
```diff
@@ -165,11 +207,8 @@ private async Task OnProcessStartedAsync()
 
             if (MeasureCpuTime)
             {
-                // Allow read access for /usr/bin/time
-                ProcessInfo.FileAccessManifest.AddPath(
-                    AbsolutePath.Create(PathTable, TimeUtil),
-                    mask: FileAccessPolicy.MaskNothing,
-                    values: FileAccessPolicy.AllowReadAlways);
+                // start the timer now
+                m_perfTimer.Change(TimeSpan.FromSeconds(0), PerfProbeInternal);
```
Comment (mpysson) [inline Public/Src/Engine/Processes/SandboxedProcess.MacKext.cs]: >m_perfTimer.Change(TimeSpan.FromSeconds(0), PerfProbeInternal); [](start = 16, length = 63)

A bit less intuitive but oftentimes the safer pattern is to only ever set the due time and not the period on the timer. Then when it fires you call Change() with a new due time in your timer callback. That prevents the timer from firing again while its still in the callback which can cause some unintended concurrency issues in the callback

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/SandboxedProcess.MacKext.cs]: will change

File: Public/Src/Engine/Processes/SandboxedProcessMac.cs [hunk 4]
```diff
@@ -136,6 +177,10 @@ protected override System.Diagnostics.Process CreateProcess()
             process.StartInfo.Arguments = string.Empty;
             process.StartInfo.RedirectStandardInput = true;
 
+            // Stop the perf timer when the process exits
+            // NOTE: even if the timer fires after the process has exited, nothing bad should happen
+            process.Exited += (o, e) => m_perfTimer.Change(Timeout.InfiniteTimeSpan, PerfProbeInternal);
```
Comment (almiliMSFT) [inline Public/Src/Engine/Processes/SandboxedProcessMac.cs]:
```suggestion
            process.Exited += (o, e) => m_perfTimer.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
```

**Pull request merged**